### ISSUE TITLE
CBD timeouts

### ISSUE THREAD

**Issue by gombaiviktor** [parodyne/esos#224]
Hello!

I'm using ESOS 1.3.9. I have VDisk based LUN on an XFS formatted soft raid array (level6). I have a lot of events like below on a Hyper-V 2012R2 server (software initiator).

Target did not respond in time for a SCSI request. The CDB is given in the dump data. (2 different CBD...)

000048000100000000000000090000C0000000000000000000000000000000000000000000000000280001034CC400000100690071006E002E0032003000310038002D00300038002E00650073006F0073002E00730074006F0072002D003300710031003A0031006500310035003700 

000044000100000000000000090000C00000000000000000000000000000000000000000000000001A000800C000690071006E002E0032003000310038002D00300038002E00650073006F0073002E00730074006F0072002D003300710031003A0031006500310035003700 

Regards,
Viktor Gombai

**Comment by msmith626** [comment]
How about the kernel logs on the target (ESOS) side when this occurs?

**Comment by gombaiviktor** [comment]
The events on the hyper-v host (initiator_1) dated: Nov  9 16:15:31

Maybe it worth to mention, that the initiator host is in a hyper-v cluster 2012R2 node with MPIO (two separate dedicated switch fabric on ethernet for SAN traffic).

I booted the ESOS host with debug mode already.

kern.log (relevant time frame):

Nov  9 16:15:11 stor-3q1 kernel: [178336.480446] [1497]: iscsi-scst: execute_task_management:2668:iSCSI TM fn 5
Nov  9 16:15:11 stor-3q1 kernel: [178336.480968] [1497]: execute_task_management:2671:TM req ffff8801a4d64000, ITT 83724002, RTT ffffffff, sn 37778051, con ffff8801fab28000
Nov  9 16:15:11 stor-3q1 kernel: [178336.481536] [1497]: scst: scst_rx_mgmt_fn:7832:TM fn LUN_RESET/4 (mcmd ffff8802223424d0, initiator iqn.1991-05.com.microsoft:##initiator_1_name_masked##, target iqn.2018-08.esos.stor-3q1:1e157)
Nov  9 16:15:11 stor-3q1 kernel: [178336.482333] [1497]: scst_rx_mgmt_fn:7846:sess=ffff8801fb239080, tag_set 0, tag 0, lun_set 1, lun=0, cmd_sn_set 1, cmd_sn 37778051, priv ffff8801a4d64000
Nov  9 16:15:11 stor-3q1 kernel: [178336.482865] [1497]: scst_post_rx_mgmt_cmd:7759:Adding mgmt cmd ffff8802223424d0 to active mgmt cmd list
Nov  9 16:15:11 stor-3q1 kernel: [178336.483472] [1480]: scst_tm_thread:7644:Deleting mgmt cmd ffff8802223424d0 from active cmd list
Nov  9 16:15:11 stor-3q1 kernel: [178336.484036] [1480]: scst: scst_lun_reset:6915:Resetting LUN 0 (mcmd ffff8802223424d0)
Nov  9 16:15:11 stor-3q1 kernel: [178336.484624] [1480]: scst: scst_alloc_set_UA:12832:Queuing new UA ffff880413a0fa90 (6:29:0, d_sense 0) to tgt_dev ffff880412138000 (dev W1_clu_data1, initiator iqn.1991-05.com.microsoft:##initiator_2_name_masked##)
Nov  9 16:15:11 stor-3q1 kernel: [178336.485531] [1480]: scst: scst_alloc_set_UA:12832:Queuing new UA ffff880413a0f430 (6:29:0, d_sense 0) to tgt_dev ffff880412158000 (dev W1_clu_data1, initiator iqn.1991-05.com.microsoft:##initiator_2_name_masked##)
Nov  9 16:15:11 stor-3q1 kernel: [178336.486342] [1480]: scst: scst_alloc_set_UA:12832:Queuing new UA ffff880413a0f320 (6:29:0, d_sense 0) to tgt_dev ffff880412158780 (dev W1_clu_data1, initiator iqn.1991-05.com.microsoft:##initiator_2_name_masked##)
Nov  9 16:15:11 stor-3q1 kernel: [178336.487148] [1480]: scst: scst_alloc_set_UA:12832:Queuing new UA ffff880413a0e880 (6:29:0, d_sense 0) to tgt_dev ffff880412158f00 (dev W1_clu_data1, initiator iqn.1991-05.com.microsoft:##initiator_2_name_masked##)
Nov  9 16:15:11 stor-3q1 kernel: [178336.487960] [1480]: scst: scst_alloc_set_UA:12832:Queuing new UA ffff880413a0e770 (6:29:0, d_sense 0) to tgt_dev ffff880412159680 (dev W1_clu_data1, initiator iqn.1991-05.com.microsoft:##initiator_2_name_masked##)
Nov  9 16:15:11 stor-3q1 kernel: [178336.488774] [1480]: scst: scst_alloc_set_UA:12832:Queuing new UA ffff880413a0e990 (6:29:0, d_sense 0) to tgt_dev ffff880412159e00 (dev W1_clu_data1, initiator iqn.1991-05.com.microsoft:##initiator_2_name_masked##)
Nov  9 16:15:11 stor-3q1 kernel: [178336.489594] [1480]: scst: scst_alloc_set_UA:12832:Queuing new UA ffff880413a0ecc0 (6:29:0, d_sense 0) to tgt_dev ffff8801fb240000 (dev W1_clu_data1, initiator iqn.1991-05.com.microsoft:##initiator_3_name_masked##)
Nov  9 16:15:11 stor-3q1 kernel: [178336.490408] [1480]: scst: scst_alloc_set_UA:12832:Queuing new UA ffff880413a0eff0 (6:29:0, d_sense 0) to tgt_dev ffff8801fb2b8f00 (dev W1_clu_data1, initiator iqn.1991-05.com.microsoft:##initiator_3_name_masked##)
Nov  9 16:15:11 stor-3q1 kernel: [178336.491225] [1480]: scst: scst_alloc_set_UA:12832:Queuing new UA ffff880413a0eee0 (6:29:0, d_sense 0) to tgt_dev ffff8801fb2b9680 (dev W1_clu_data1, initiator iqn.1991-05.com.microsoft:##initiator_3_name_masked##)
Nov  9 16:15:11 stor-3q1 kernel: [178336.492036] [1480]: scst: scst_alloc_set_UA:12832:Queuing new UA ffff880413a0edd0 (6:29:0, d_sense 0) to tgt_dev ffff8801fb240780 (dev W1_clu_data1, initiator iqn.1991-05.com.microsoft:##initiator_1_name_masked##)
Nov  9 16:15:11 stor-3q1 kernel: [178336.492850] [1480]: scst: scst_alloc_set_UA:12832:Queuing new UA ffff880413a0ebb0 (6:29:0, d_sense 0) to tgt_dev ffff880412138780 (dev W1_clu_data1, initiator iqn.1991-05.com.microsoft:##initiator_1_name_masked##)
Nov  9 16:15:11 stor-3q1 kernel: [178336.493669] [1480]: scst: scst_alloc_set_UA:12832:Queuing new UA ffff880413a0e000 (6:29:0, d_sense 0) to tgt_dev ffff880412138f00 (dev W1_clu_data1, initiator iqn.1991-05.com.microsoft:##initiator_1_name_masked##)
Nov  9 16:15:11 stor-3q1 kernel: [178336.500562] [1480]: scst: scst_alloc_set_UA:12832:Queuing new UA ffff880413a0e660 (6:29:0, d_sense 0) to tgt_dev ffff88041215a580 (dev W1_clu_data1, initiator iqn.1991-05.com.microsoft:##initiator_1_name_masked##)
Nov  9 16:15:11 stor-3q1 kernel: [178336.501380] [1480]: scst: scst_alloc_set_UA:12832:Queuing new UA ffff880413a0e110 (6:29:0, d_sense 0) to tgt_dev ffff8801fb240f00 (dev W1_clu_data1, initiator iqn.1991-05.com.microsoft:##initiator_1_name_masked##)
Nov  9 16:15:11 stor-3q1 kernel: [178336.502186] [1480]: scst: scst_alloc_set_UA:12832:Queuing new UA ffff880413a0e550 (6:29:0, d_sense 0) to tgt_dev ffff8801fb241680 (dev W1_clu_data1, initiator iqn.1991-05.com.microsoft:##initiator_1_name_masked##)
Nov  9 16:15:11 stor-3q1 kernel: [178336.503001] [1480]: scst: scst_alloc_set_UA:12832:Queuing new UA ffff880413a0e440 (6:29:0, d_sense 0) to tgt_dev ffff88041215b480 (dev W1_clu_data1, initiator iqn.1991-05.com.microsoft:##initiator_3_name_masked##)
Nov  9 16:15:11 stor-3q1 kernel: [178336.503819] [1480]: scst: scst_alloc_set_UA:12832:Queuing new UA ffff880413a0e330 (6:29:0, d_sense 0) to tgt_dev ffff8801fb7df800 (dev W1_clu_data1, initiator iqn.1991-05.com.microsoft:##initiator_3_name_masked##)
Nov  9 16:15:11 stor-3q1 kernel: [178336.504634] [1480]: scst: scst_alloc_set_UA:12832:Queuing new UA ffff880413a0e220 (6:29:0, d_sense 0) to tgt_dev ffff8801e9602d00 (dev W1_clu_data1, initiator iqn.1991-05.com.microsoft:##initiator_3_name_masked##)
Nov  9 16:15:11 stor-3q1 kernel: [178336.505460] [1480]: scst_call_dev_task_mgmt_fn_done:6135:Calling dev handler vdisk_fileio task_mgmt_fn_done(fn=4)
Nov  9 16:15:11 stor-3q1 kernel: [178336.505998] [1480]: scst_call_dev_task_mgmt_fn_done:6138:Dev handler vdisk_fileio task_mgmt_fn_done() returned
Nov  9 16:15:11 stor-3q1 kernel: [178336.506543] [1480]: iscsi_task_mgmt_affected_cmds_done:332:scst_mcmd ffff8802223424d0, fn 4, priv ffff8801a4d64000
Nov  9 16:15:11 stor-3q1 kernel: [178336.507079] [1480]: scst_call_task_mgmt_affected_cmds_done:7358:Target's iscsi task_mgmt_affected_cmds_done() returned
Nov  9 16:15:11 stor-3q1 kernel: [178336.507629] [1480]: scst: scst_mgmt_cmd_send_done:7495:TM fn 4 (mcmd ffff8802223424d0) finished, status 0
Nov  9 16:15:11 stor-3q1 kernel: [178336.508168] [1480]: iscsi_task_mgmt_fn_done:3797:req ffff8801a4d64000, scst_mcmd ffff8802223424d0, fn 4, scst status 0, status 0
Nov  9 16:15:11 stor-3q1 kernel: [178336.508715] [1480]: iscsi_send_task_mgmt_resp:3687:TM req ffff8801a4d64000 finished
Nov  9 16:15:11 stor-3q1 kernel: [178336.509253] [1480]: iscsi-scst: iscsi_send_task_mgmt_resp:3689:iSCSI TM fn 5 finished, status 0, dropped 0
Nov  9 16:15:11 stor-3q1 kernel: [178336.509792] [1480]: scst_mgmt_cmd_send_done:7510:Target's iscsi task_mgmt_fn_done() returned
Nov  9 16:15:11 stor-3q1 kernel: [178336.510328] [1527]: iscsi_get_send_cmnd:684:Going to send TM response ffff88018d076e80 (status 0, fn 5, parent_req ffff8801a4d64000)
Nov  9 16:15:11 stor-3q1 kernel: [178336.510863] [1840]: scst_set_pending_UA:12687:Setting pending UA cmd ffff8800c7dbf0c0 (tgt_dev ffff880412138780, dev W1_clu_data1, initiator iqn.1991-05.com.microsoft:##initiator_1_name_masked##)
Nov  9 16:15:12 stor-3q1 kernel: [178337.511527] [1833]: scst_set_pending_UA:12687:Setting pending UA cmd ffff8800c7dbe5c0 (tgt_dev ffff8801fb240780, dev W1_clu_data1, initiator iqn.1991-05.com.microsoft:##initiator_1_name_masked##)
Nov  9 16:15:12 stor-3q1 kernel: [178337.695132] [1806]: scst_set_pending_UA:12687:Setting pending UA cmd ffff8800c7dbb9c0 (tgt_dev ffff8801fb2b8f00, dev W1_clu_data1, initiator iqn.1991-05.com.microsoft:##initiator_3_name_masked##)
Nov  9 16:15:13 stor-3q1 kernel: [178338.512643] [1851]: scst_set_pending_UA:12687:Setting pending UA cmd ffff8800690a82c0 (tgt_dev ffff880412138f00, dev W1_clu_data1, initiator iqn.1991-05.com.microsoft:##initiator_1_name_masked##)
Nov  9 16:15:13 stor-3q1 kernel: [178338.531268] [1816]: scst_set_pending_UA:12687:Setting pending UA cmd ffff8800690a9b80 (tgt_dev ffff8801fb2b9680, dev W1_clu_data1, initiator iqn.1991-05.com.microsoft:##initiator_3_name_masked##)
Nov  9 16:15:13 stor-3q1 kernel: [178338.532085] [1785]: scst_set_pending_UA:12687:Setting pending UA cmd ffff8800c7db9e40 (tgt_dev ffff8801fb240000, dev W1_clu_data1, initiator iqn.1991-05.com.microsoft:##initiator_3_name_masked##)
Nov  9 16:15:14 stor-3q1 kernel: [178339.399804] [31753]: scst_set_pending_UA:12687:Setting pending UA cmd ffff8800c7db9600 (tgt_dev ffff8801e9602d00, dev W1_clu_data1, initiator iqn.1991-05.com.microsoft:##initiator_3_name_masked##)
Nov  9 16:15:14 stor-3q1 kernel: [178339.400770] [31623]: scst_set_pending_UA:12687:Setting pending UA cmd ffff8800690ab9c0 (tgt_dev ffff88041215b480, dev W1_clu_data1, initiator iqn.1991-05.com.microsoft:##initiator_3_name_masked##)
Nov  9 16:15:14 stor-3q1 kernel: [178339.513648] [1865]: scst_set_pending_UA:12687:Setting pending UA cmd ffff8800690ac780 (tgt_dev ffff88041215a580, dev W1_clu_data1, initiator iqn.1991-05.com.microsoft:##initiator_1_name_masked##)
Nov  9 16:15:14 stor-3q1 kernel: [178339.523229] [31733]: scst_set_pending_UA:12687:Setting pending UA cmd ffff8800c7dbbc80 (tgt_dev ffff8801fb7df800, dev W1_clu_data1, initiator iqn.1991-05.com.microsoft:##initiator_3_name_masked##)
Nov  9 16:15:16 stor-3q1 kernel: [178341.997316] [1868]: scst_set_pending_UA:12687:Setting pending UA cmd ffff8800690ac200 (tgt_dev ffff8801fb240f00, dev W1_clu_data1, initiator iqn.1991-05.com.microsoft:##initiator_1_name_masked##)
Nov  9 16:15:31 stor-3q1 kernel: [178356.424701] [0]: iscsi-scst: __iscsi_state_change:502:***ERROR***: Connection ffff8801fab28000 with initiator iqn.1991-05.com.microsoft:##initiator_1_name_masked## unexpectedly closed!
Nov  9 16:15:31 stor-3q1 kernel: [178356.425500] [0]: __iscsi_state_change:504:conn ffff8801fab28000, sk state 7
Nov  9 16:15:31 stor-3q1 kernel: [178356.426075] [34028]: close_conn:411:Closing connection ffff8801fab28000 (conn_ref_cnt=1)
Nov  9 16:15:31 stor-3q1 kernel: [178356.426605] [34028]: scst: scst_rx_mgmt_fn:7832:TM fn NEXUS_LOSS_SESS/6 (mcmd ffff8801fbb65690, initiator iqn.1991-05.com.microsoft:##initiator_1_name_masked##, target iqn.2018-08.esos.stor-3q1:1e157)
Nov  9 16:15:31 stor-3q1 kernel: [178356.427402] [1544]: __add_session:169:Adding session SID 55050037010040
Nov  9 16:15:31 stor-3q1 kernel: [178356.427685] [34028]: scst_rx_mgmt_fn:7846:sess=ffff8801fb239080, tag_set 0, tag 0, lun_set 1, lun=0, cmd_sn_set 0, cmd_sn 0, priv ffff8801fab28000
Nov  9 16:15:31 stor-3q1 kernel: [178356.428220] [1544]: scst: scst_init_session:8136:Using security group "SRHV-W1" for initiator "iqn.1991-05.com.microsoft:##initiator_1_name_masked##" (target iqn.2018-08.esos.stor-3q1:1e157)
Nov  9 16:15:31 stor-3q1 kernel: [178356.429005] [34028]: scst_post_rx_mgmt_cmd:7759:Adding mgmt cmd ffff8801fbb65690 to active mgmt cmd list
Nov  9 16:15:31 stor-3q1 kernel: [178356.429530] [1544]: scst_alloc_add_tgt_dev:5315:Device W1_clu_data1 on SCST lun=0
Nov  9 16:15:31 stor-3q1 kernel: [178356.430140] [34028]: req_cmnd_release_force:600:req ffff8801a4d67000
Nov  9 16:15:31 stor-3q1 kernel: [178356.430412] [1480]: scst_tm_thread:7644:Deleting mgmt cmd ffff8801fbb65690 from active cmd list
Nov  9 16:15:31 stor-3q1 kernel: [178356.430950] [1544]: scst: scst_alloc_set_UA:12832:Queuing new UA ffff8801fb29d210 (6:29:0, d_sense 0) to tgt_dev ffff8801fb2bda00 (dev W1_clu_data1, initiator iqn.1991-05.com.microsoft:##initiator_1_name_masked##)
Nov  9 16:15:31 stor-3q1 kernel: [178356.431738] [34028]: conn_abort:2597:Aborting conn ffff8801fab28000
Nov  9 16:15:31 stor-3q1 kernel: [178356.432011] [1480]: scst_abort_all_nexus_loss_sess:6998:Nexus loss or UNREG SESS for sess ffff8801fb239080 (mcmd ffff8801fbb65690)
Nov  9 16:15:31 stor-3q1 kernel: [178356.432561] [34029]: scst: scst_cmd_thread:5654:Processing thread W1_clu_dat20_0 started
Nov  9 16:15:31 stor-3q1 kernel: [178356.433090] [34030]: scst: scst_cmd_thread:5654:Processing thread W1_clu_dat20_1 started
Nov  9 16:15:31 stor-3q1 kernel: [178356.433629] [34031]: scst: scst_cmd_thread:5654:Processing thread W1_clu_dat20_2 started
Nov  9 16:15:31 stor-3q1 kernel: [178356.434156] [34032]: scst: scst_cmd_thread:5654:Processing thread W1_clu_dat20_3 started
Nov  9 16:15:31 stor-3q1 kernel: [178356.434698] [34033]: scst: scst_cmd_thread:5654:Processing thread W1_clu_dat20_4 started
Nov  9 16:15:31 stor-3q1 kernel: [178356.435226] [34034]: scst: scst_cmd_thread:5654:Processing thread W1_clu_dat20_5 started
Nov  9 16:15:31 stor-3q1 kernel: [178356.435768] [34035]: scst: scst_cmd_thread:5654:Processing thread W1_clu_dat20_6 started
Nov  9 16:15:31 stor-3q1 kernel: [178356.436304] [34036]: scst: scst_cmd_thread:5654:Processing thread W1_clu_dat20_7 started
Nov  9 16:15:31 stor-3q1 kernel: [178356.436856] [1544]: iscsi-scst: iscsi_session_alloc:118:Session ffff8801fb2ae900 created: target ffff8802262a7e00, tid 1, sid 0x55050037010040, initiator iqn.1991-05.com.microsoft:##initiator_1_name_masked##
Nov  9 16:15:31 stor-3q1 kernel: [178356.437651] [1480]: scst: scst_alloc_set_UA:12832:Queuing new UA ffff880413a0f100 (6:29:7, d_sense 0) to tgt_dev ffff8801fb241680 (dev W1_clu_data1, initiator iqn.1991-05.com.microsoft:##initiator_1_name_masked##)
Nov  9 16:15:31 stor-3q1 kernel: [178356.438475] [1544]: iscsi-scst: log_params:90:Negotiated parameters: InitialR2T No, ImmediateData Yes, MaxConnections 1, MaxRecvDataSegmentLength 1048576, MaxXmitDataSegmentLength 65536,
Nov  9 16:15:31 stor-3q1 kernel: [178356.439261] [1480]: scst_call_dev_task_mgmt_fn_done:6135:Calling dev handler vdisk_fileio task_mgmt_fn_done(fn=6)
Nov  9 16:15:31 stor-3q1 kernel: [178356.439806] [1544]: iscsi-scst: log_params:94:    MaxBurstLength 262144, FirstBurstLength 65536, DefaultTime2Wait 0, DefaultTime2Retain 0,
Nov  9 16:15:31 stor-3q1 kernel: [178356.440337] [1480]: scst_call_dev_task_mgmt_fn_done:6138:Dev handler vdisk_fileio task_mgmt_fn_done() returned
Nov  9 16:15:31 stor-3q1 kernel: [178356.440881] [1544]: iscsi-scst: log_params:100:    MaxOutstandingR2T 16, DataPDUInOrder Yes, DataSequenceInOrder Yes, ErrorRecoveryLevel 0,
Nov  9 16:15:31 stor-3q1 kernel: [178356.441411] [1480]: iscsi_task_mgmt_affected_cmds_done:332:scst_mcmd ffff8801fbb65690, fn 6, priv ffff8801fab28000
Nov  9 16:15:31 stor-3q1 kernel: [178356.441953] [1544]: iscsi-scst: log_params:108:    HeaderDigest None, DataDigest None, OFMarker No, IFMarker No, OFMarkInt 2048, IFMarkInt 2048, RDMAExtensions No
Nov  9 16:15:31 stor-3q1 kernel: [178356.442745] [1544]: iscsi-scst: iscsi_tgt_params_set:277:Target parameters set for session 55050037010040: QueuedCommands 32, Response timeout 90, Nop-In interval 30, Nop-In timeout 30
Nov  9 16:15:31 stor-3q1 kernel: [178356.443540] [1544]: __add_session:251:Reinstating sess ffff8801fb2ae900 with SID 55050037010040 (old ffff8801fb233480, SID 13050037010040)
Nov  9 16:15:31 stor-3q1 kernel: [178356.444075] [1544]: iscsi-scst: target_del_session:297:Deleting session ffff8801fb233480 (initiator iqn.1991-05.com.microsoft:##initiator_1_name_masked##)
Nov  9 16:15:31 stor-3q1 kernel: [178356.444686] [1544]: target_del_session:306:Del session: closing conn ffff8801fab28000
Nov  9 16:15:31 stor-3q1 kernel: [178356.445211] [1544]: scst_set_initial_UA:2441:Setting for sess ffff8801fb2b4200 initial UA 6/29/7
Nov  9 16:15:31 stor-3q1 kernel: [178356.445738] [1480]: sess_reinst_finished:144:Enabling reinstate successor sess ffff8801fb2ae900
Nov  9 16:15:31 stor-3q1 kernel: [178356.446308] [1480]: scst_call_task_mgmt_affected_cmds_done:7358:Target's iscsi task_mgmt_affected_cmds_done() returned
Nov  9 16:15:31 stor-3q1 kernel: [178356.446852] [1544]: iscsi-scst: iscsi_conn_alloc:1004:Creating connection ffff8801fb272ac0 for sid 0x55050037010040, cid 1 (initiator iqn.1991-05.com.microsoft:##initiator_1_name_masked##)
Nov  9 16:15:31 stor-3q1 kernel: [178356.447635] [1480]: scst: scst_mgmt_cmd_send_done:7495:TM fn 6 (mcmd ffff8801fbb65690) finished, status 0
Nov  9 16:15:31 stor-3q1 kernel: [178356.448175] [1544]: conn_activate:800:Enabling conn ffff8801fb272ac0
Nov  9 16:15:31 stor-3q1 kernel: [178356.448447] [1480]: iscsi_task_mgmt_fn_done:3797:req ffff8801fab28000, scst_mcmd ffff8801fbb65690, fn 6, scst status 0, status 0
Nov  9 16:15:31 stor-3q1 kernel: [178356.448996] [34036]: scst_set_pending_UA:12687:Setting pending UA cmd ffff8800690af640 (tgt_dev ffff8801fb2bda00, dev W1_clu_data1, initiator iqn.1991-05.com.microsoft:##initiator_1_name_masked##)
Nov  9 16:15:31 stor-3q1 kernel: [178356.449823] [34028]: iscsi-scst: conn_free:886:Freeing conn ffff8801fab28000 (sess=ffff8801fb233480, 0x13050037010040 1, initiator iqn.1991-05.com.microsoft:##initiator_1_name_masked##)
Nov  9 16:15:31 stor-3q1 kernel: [178356.450616] [1480]: scst_mgmt_cmd_send_done:7510:Target's iscsi task_mgmt_fn_done() returned
Nov  9 16:15:31 stor-3q1 kernel: [178356.451157] [34036]: scst: scst_set_cmd_error_status:1796:Reservation conflict (dev W1_clu_data1, initiator iqn.1991-05.com.microsoft:##initiator_1_name_masked##, tgt_id 1)
Nov  9 16:15:31 stor-3q1 kernel: [178356.451971] [34028]: iscsi-scst: session_free:323:Freeing session ffff8801fb233480 (SID 13050037010040)
Nov  9 16:15:31 stor-3q1 kernel: [178356.452496] [34036]: scst: scst_set_cmd_error_status:1796:Reservation conflict (dev W1_clu_data1, initiator iqn.1991-05.com.microsoft:##initiator_1_name_masked##, tgt_id 1)
Nov  9 16:15:31 stor-3q1 kernel: [178356.453312] [34028]: scst_unregister_session:8383:Unregistering session ffff8801fb239080 (wait 0)
Nov  9 16:15:31 stor-3q1 kernel: [178356.453848] [34028]: scst_rx_mgmt_fn:7836:TM fn UNREG_SESS_TM/10 (mcmd ffff8801fbb65700)
Nov  9 16:15:31 stor-3q1 kernel: [178356.454370] [34036]: scst: scst_set_cmd_error_status:1796:Reservation conflict (dev W1_clu_data1, initiator iqn.1991-05.com.microsoft:##initiator_1_name_masked##, tgt_id 1)
Nov  9 16:15:31 stor-3q1 kernel: [178356.455177] [34028]: scst_rx_mgmt_fn:7846:sess=ffff8801fb239080, tag_set 0, tag 0, lun_set 0, lun=0, cmd_sn_set 0, cmd_sn 0, priv           (null)
Nov  9 16:15:31 stor-3q1 kernel: [178356.455714] [34036]: scst: scst_set_cmd_error_status:1796:Reservation conflict (dev W1_clu_data1, initiator iqn.1991-05.com.microsoft:##initiator_1_name_masked##, tgt_id 1)
Nov  9 16:15:31 stor-3q1 kernel: [178356.456528] [34028]: scst_post_rx_mgmt_cmd:7759:Adding mgmt cmd ffff8801fbb65700 to active mgmt cmd list
Nov  9 16:15:31 stor-3q1 kernel: [178356.457084] [34028]: close_conn:578:Notifying user space about closing connection ffff8801fab28000
Nov  9 16:15:31 stor-3q1 kernel: [178356.457618] [1480]: scst_tm_thread:7644:Deleting mgmt cmd ffff8801fbb65700 from active cmd list
Nov  9 16:15:31 stor-3q1 kernel: [178356.458161] [34034]: scst: scst_set_cmd_error_status:1796:Reservation conflict (dev W1_clu_data1, initiator iqn.1991-05.com.microsoft:##initiator_1_name_masked##, tgt_id 1)
Nov  9 16:15:31 stor-3q1 kernel: [178356.459058] [34035]: scst: scst_set_cmd_error_status:1796:Reservation conflict (dev W1_clu_data1, initiator iqn.1991-05.com.microsoft:##initiator_1_name_masked##, tgt_id 1)
Nov  9 16:15:31 stor-3q1 kernel: [178356.459841] [1480]: scst_abort_all_nexus_loss_sess:6998:Nexus loss or UNREG SESS for sess ffff8801fb239080 (mcmd ffff8801fbb65700)
Nov  9 16:15:31 stor-3q1 kernel: [178356.460414] [1480]: scst_call_dev_task_mgmt_fn_done:6135:Calling dev handler vdisk_fileio task_mgmt_fn_done(fn=10)
Nov  9 16:15:31 stor-3q1 kernel: [178356.460967] [1480]: scst_call_dev_task_mgmt_fn_done:6138:Dev handler vdisk_fileio task_mgmt_fn_done() returned
Nov  9 16:15:31 stor-3q1 kernel: [178356.461509] [1480]: scst_mgmt_cmd_send_done:7498:TM fn 10 (mcmd ffff8801fbb65700) finished, status 0
Nov  9 16:15:31 stor-3q1 kernel: [178356.462153] [1481]: scst_free_all_UA:12988:Clearing UA for tgt_dev LUN 0
Nov  9 16:15:31 stor-3q1 kernel: [178356.462435] [1481]: scst_free_all_UA:12988:Clearing UA for tgt_dev LUN 0
Nov  9 16:15:31 stor-3q1 kernel: [178356.462791] [2065]: scst: scst_cmd_thread:5833:Processing thread W1_clu_dat19_0 finished
Nov  9 16:15:31 stor-3q1 kernel: [178356.463411] [2066]: scst: scst_cmd_thread:5833:Processing thread W1_clu_dat19_1 finished
Nov  9 16:15:31 stor-3q1 kernel: [178356.463995] [2067]: scst: scst_cmd_thread:5833:Processing thread W1_clu_dat19_2 finished
Nov  9 16:15:31 stor-3q1 kernel: [178356.464590] [2068]: scst: scst_cmd_thread:5833:Processing thread W1_clu_dat19_3 finished
Nov  9 16:15:31 stor-3q1 kernel: [178356.465208] [2069]: scst: scst_cmd_thread:5833:Processing thread W1_clu_dat19_4 finished
Nov  9 16:15:31 stor-3q1 kernel: [178356.465797] [2070]: scst: scst_cmd_thread:5833:Processing thread W1_clu_dat19_5 finished
Nov  9 16:15:31 stor-3q1 kernel: [178356.466395] [2071]: scst: scst_cmd_thread:5833:Processing thread W1_clu_dat19_6 finished
Nov  9 16:15:31 stor-3q1 kernel: [178356.466967] [2072]: scst: scst_cmd_thread:5833:Processing thread W1_clu_dat19_7 finished
Nov  9 16:42:53 stor-3q1 kernel: [179998.452737] [1489]: iscsi-scst: execute_task_management:2668:iSCSI TM fn 5

**Comment by msmith626** [comment]
I see messages regarding reservation conflicts, but this may just be
"normal" or may mean nothing and we only see those messages because debug
is enabled. I see session disconnect messages as well.. maybe some type of
network connectivity issues between both sides? Or mis-matched MTU between
both sides, or something along those lines.

--Marc

**Comment by gombaiviktor** [comment]
I have checked the SAN network and I found no issues (we have another
iscsi storage (NetApp) as well and it works on the same switching fabric
without problems. We have two independent switches, the target and the
clients have 4 NICs with 4 different IP network for MPIO). The MTU is the
same on every interfaces: 1500 (I know, it would be better to have it 9000).

When the reservation conflicts appear in the log, the performance drops and
session recovery occurs on the initiators (cluster drops the target for a
moment). The problem happens very likely when a vm storage migration or a
vm boot happens on the hypervisor.

The cluster event is: eventid:5120  "Cluster Shared Volume 'Volume8'
('3q1-W1-data1') has entered a paused state because of '(80000011)'. All
I/O will temporarily be queued until a path to the volume is
reestablished."

It seems simple because 80000011 is about "device is busy", but no
significant load or IO wait on the target at all...

Viktor

2018. nov. 11., V 17:33 dátummal Marc A. Smith <notifications@github.com>
ezt írta:

**Comment by msmith626** [comment]
Hi,

I would need to access the systems directly to diagnose further. A packet
capture using something like Wireshark may also be beneficial. This type of
support is out of scope for the OSS license. Commercial support options for
ESOS are available from Parodyne if you're interested. Posting your logs
and preliminary diagnostic steps on the "esos-users" mailing list may be
worthwhile as others may have overcome these issues with a similar setup.

--Marc
